main.go

```diff
@@ -903,11 +902,10 @@ func loadConfig(configFilePath string, logger log.Logger) (*config.GatewayConfig
 	flags.addFlagsToConfig(loadedConfig)
 
 	// log the config that we loaded
-	var bb []byte
-	if bb, err = json.Marshal(loadedConfig); err == nil {
-		logger.Log(logkey.Config, string(bb), logkey.Env, strings.Join(os.Environ(), "-"), "config loaded")
+	if _, err = json.Marshal(loadedConfig); err != nil {
```

Comment:
If we're not dumping the config anymore it might not be necessary to re-marshal it.

We should still offer a mechanism to preserve the previous behavior, for example through a command-line flag that asks the gateway to log its config, after calling `loadConfig()` in the `main()` method:

```go
loadedConfig, err := loadConfig(flags.configFileName, logger)
if &flags.showConfig {
  ... log the marshalled loadedConfig ...
}
```

So that one can run `gateway --show-config` or something like that.

Reply:
I wasn't paying too much attention what was happening within this section of code, and I assumed it was an additional validation step after loading all the variables from their various sources. 

I am really sceptical about adding any functionality that has the potential to leak secrets and would much rather the config to be manually inspected. However, knowing that some of the config is partially loaded from the environment, perhaps some info logs regarding that it has loaded info from the environment.